### PR TITLE
Initial OIIO 2 support

### PR DESCRIPTION
**Pull request by luisbarrancos**
ImageInput and ImageOutput static create() and open() methods now return
unique_ptr rather than raw pointers, and OpenVDB is now read via texture3d(). It also brings us the gradients needed for bump2roughness. Notes at https://github.com/OpenImageIO/oiio/releases/tag/Release-2.0.3

### CODE REVIEW

File: src/appleseed/foundation/image/genericimagefilewriter.h [hunk 1]
```diff
@@ -92,7 +93,12 @@ class APPLESEED_DLLSYMBOL GenericImageFileWriter
     struct Impl;
     Impl* impl;
 
+#if OIIO_VERSION >= 20000
+    std::unique_ptr<OIIO::ImageOutput> m_writer;
```
Comment (dictoon) [inline src/appleseed/foundation/image/genericimagefilewriter.h]: You need to include `<memory>` for the definition of `std::unique_ptr<>`.

Comment (dictoon) [reply src/appleseed/foundation/image/genericimagefilewriter.h]: Also, can we move `m_writer` and `m_filename` to the private impl?

File: src/appleseed/foundation/image/genericprogressiveimagefilereader.cpp [hunk 2]
```diff
@@ -177,7 +184,9 @@ void GenericProgressiveImageFileReader::close()
     // but OpenImageIO 1.5.20 (the version included in appleseed-deps at the time
     // of writing) is too old to have this method. Since on Windows we link to
     // OpenImageIO statically, this should be safe anyway.
+#if OIIO_VERSION < 20000
```
Comment (dictoon) [inline src/appleseed/foundation/image/genericprogressiveimagefilereader.cpp]: Not strictly part of your PR, but... Is the comment above still relevant? If it is and we keep the comment, then it should be inside the `#if ... #endif` block.

Comment (luisbarrancos) [reply src/appleseed/foundation/image/genericprogressiveimagefilereader.cpp]: Not relevant anymore, and OIIO::ImageInput::destroy() can be called.

File: src/appleseed/foundation/image/genericimagefilewriter.h (hunk 1 above)
Comment (luisbarrancos) [reply src/appleseed/foundation/image/genericimagefilewriter.h]: Done, but still testing.

File: src/appleseed/foundation/image/genericimagefilewriter.cpp [hunk 3]
```diff
@@ -57,6 +60,13 @@ struct GenericImageFileWriter::Impl
 {
     std::vector<const ICanvas*>     m_canvas;
     std::vector<OIIO::ImageSpec>    m_spec;
+
+#ifdef OIIO_VERSION >= 20000
+    std::unique_ptr<OIIO::ImageOutput>  m_writer;
```
Comment (dictoon) [inline src/appleseed/foundation/image/genericimagefilewriter.cpp]: Align members.

Comment (dictoon) [reply src/appleseed/foundation/image/genericimagefilewriter.cpp]: I would also suggest ordering them similarly to how they are initialized, which seems to be: filename, writer, canvas, spec.

File: src/appleseed/foundation/image/genericimagefilewriter.cpp [hunk 4]
```diff
@@ -320,6 +331,7 @@ void GenericImageFileWriter::write_tiles(const size_t image_index)
     // Compute the tiles' xstride offset in bytes.
     size_t xstride = props.m_pixel_size;
 
+
```
Comment (dictoon) [inline src/appleseed/foundation/image/genericimagefilewriter.cpp]: Remove redundant blank line.

File: src/appleseed/foundation/image/genericimagefilewriter.h [hunk 5]
```diff
@@ -38,8 +38,12 @@
 // OpenImageIO headers.
 #include "foundation/platform/_beginoiioheaders.h"
 #include "OpenImageIO/imageio.h"
+#include "OpenImageIO/version.h"
 #include "foundation/platform/_endoiioheaders.h"
 
+// Standard headers.
```
Comment (dictoon) [inline src/appleseed/foundation/image/genericimagefilewriter.h]: Remove dead code.

**Pull request merged**